src/App.vue

```diff
@@ -18,16 +18,23 @@
 </template>
 
 <script lang="ts">
-import { Vue, Component } from 'vue-property-decorator';
+import { Component, Mixins } from 'vue-property-decorator';
 import { auth } from '@/firebase';
+
+import Perf from '@/mixins/Perf.ts';
 import Toolbar from '@/components/Toolbar.vue';
 
 @Component({
   components: { Toolbar },
 })
-export default class App extends Vue {
+export default class App extends Mixins(Perf) {
   signedIn() {
     return !!auth.currentUser;
   }
+  mounted() {
+    this.$nextTick(() => {
+      this.logReady('app_loaded', { userAgent: navigator.userAgent });
```

Comment:
Should the string constants "app_loaded" etc be in a single file?

Reply:
no, i don't think so. in my experience, it's usually better to store identifiers used in log or error messages in the code that produces those messages, since it makes it much easier to find the responsible code when looking at logs.

for the same reason, i find that it's often best to use literal strings in log messages instead of refactoring common code into a shared location: it's impossible to grep the code for a string that's being dynamically generated.

(don't get me started on python code that uses `__getattr__` to dynamically create attributes as they're accessed, making it impossible to grep for the responsible code while you're looking at a caller...)